### DISCUSSION TITLE
How to display the "Open Changes With Next Revision" button in the top right tab bar?

### DISCUSSION THREAD

**Discussion by ale-diez** [gitkraken/vscode-gitlens#2264] [Q&A] [answered]
When working with a file I can correctly see the all the three GitLens options, however when navigation to a previous revision I don't see the button to go to the next revision on the toolbar and I have to open the dropdown menu to navigate to it

Working on a file
![image](https://user-images.githubusercontent.com/16702156/195601280-10356a63-2a2f-4771-82f5-e8cd323856f2.png)

After switching to a previous revision; the option to navigate to the next revision is hidden in the dropdown menu
![image](https://user-images.githubusercontent.com/16702156/195601853-e065c8b2-bcae-4e14-a0f0-a557673ac2f9.png)

Is it possible to put it back in the tab bar?

**Comment by LiveFreeAndRoam** [comment]
This is an old question, but I am seeing this same beahviour on a Windows installation of VSCode.  Another odd behaviour is that when I go back one revision, the "Revision Navigation" menu is completely removed.  E.g.

Open a 'C' file and the menu looks like this:

![image](https://github.com/gitkraken/vscode-gitlens/assets/8231064/66f403f7-9ee3-4b1c-aa24-57f80a89e214)


Click the "Previous Revision" icon and the toolbar changes to:

![image](https://github.com/gitkraken/vscode-gitlens/assets/8231064/68ed7115-e03f-4647-816f-9a414c06843c)

Where the icons have been moved into the drop-down menu:

![image](https://github.com/gitkraken/vscode-gitlens/assets/8231064/0573f2e0-5649-4730-a7da-aa561deeaaff)

Is there some limit to the number of icons that can be displayed? How can I increase that number, or configure it so the Git Lens Icons are always displayed?

**Comment by jkelroy** [comment, marked as answer]
Hello @demedos @LiveFreeAndRoam, this is a limitation/restriction of VS Code as it only allows a certain number of icons in the editor toolbar before they will be moved to the `...` icon. If you would like to be able to pin specific actions or see more in the editor toolbar, I would recommend submitting a feature request to the [VS Code repository](https://github.com/microsoft/vscode/issues/new/choose).

Edit: the buttons in this menu can be hidden by right-clicking them, which will make room for GitLens buttons if desired.

**Comment by LiveFreeAndRoam** [comment]
Thanks @jkelroy .  I have created an enhancement request https://github.com/microsoft/vscode/issues/182802.

Hopefully, it will get enough upvotes or the designers will see the need for this regardless.  The clock has started.
